paltalabs/create-soroban-dapp
# fixed the connector issue

# Fix xBull Wallet Popup Issue When Signing with Other Wallets
# Fixes #60 
## Problem
When signing a transaction with wallets other than xBull (e.g., Freighter), the xBull wallet popup would unexpectedly appear. This caused confusion and a poor user experience.

## Solution
We've updated the `GreeterContractInteractions.tsx` file to ensure that only the actively selected wallet is used for signing transactions. This change prevents the xBull wallet from interfering when it's not the chosen wallet.

## Changes Made
1. Updated the `updateGreeting` function to use the `activeConnector` for signing transactions.
2. Removed dependencies on specific wallet implementations, making the code more wallet-agnostic.
3. Added error handling and improved transaction status checking.

## Specific Changes
```typescript
const updateGreeting = async ({ newMessage }: UpdateGreetingValues) => {
  if (!address || !server || !activeChain || !activeConnector) {
    toast.error('Wallet not connected or missing configuration. Try again...');
    return;
  }

  setUpdateIsLoading(true);

  try {
    const account = await server.getAccount(address);
    
    const contract = new StellarSdk.Contract(contractAddressStored || '');
    const tx = new StellarSdk.TransactionBuilder(account, { 
      fee: StellarSdk.BASE_FEE,
      networkPassphrase: activeChain.networkPassphrase
    })
      .addOperation(contract.call("set_title", nativeToScVal(newMessage, {type: "string"})))
      .setTimeout(30)
      .build();

    const simulation = await server.simulateTransaction(tx);
    
    if ('results' in simulation) {
      const preparedTransaction = tx;

      const signedTransaction = await activeConnector.signTransaction(
        preparedTransaction.toXDR(),
        { networkPassphrase: activeChain.networkPassphrase }
      );
      
      // ... (rest of the function)
    }
  } catch (e) {
    console.error(e);
    toast.error('Error while sending tx. Try again…');
  } finally {
    setUpdateIsLoading(false);
  }
};
```

## Testing
To test this fix:
1. Connect different wallets (Freighter, Lobstr, Hana, xBull) one at a time.
2. Attempt to change the greeting using each connected wallet.
3. Verify that only the connected wallet's popup appears for signing.
4. Ensure the greeting can be successfully changed with each wallet.

## Additional Notes
This PR is part of the fix for issue #60. It addresses the frontend component of the problem. A corresponding update to the wallet initialization in `MySorobanReactProvider.tsx` may be necessary for a complete solution.
